4SUM Solution (#1637)
cpinitiative/usaco-guide: solutions/cses-1642.mdx

---
id: cses-1642
source: CSES
title: Sum of Four Values
author: Neo Wang
---

## Explanation

Since we are looking for four numbers $a + b + c + d = x$, we can store all the

> values that we can achieve, and the two numbers that add up to

Reviewer: > all the values that we can achieve

What do you mean by this?

> values that we can achieve, and the two numbers that add up to
> that value using a map. Then, the problem is reduced to finding two numbers $a$

Author: ```suggestion
values that we can achieve using a pair of numbers, and the indices for both numbers in the pair using a map. Then, the problem is reduced to finding two numbers $a$
```

> and $b$ that sum up to a mapped value such that no indice of $a, b, c, d$ is repeated.

Reviewer: ```suggestion
and $b$ that sum up to a mapped value such that no index is repeated.
```
